core-android/src/main/java/com/uber/sdk/android/core/auth/LoginManager.java

```diff
@@ -363,4 +410,54 @@ private void handleResultOk(@Nullable Intent data) {
 
         }
     }
+
+    boolean isInLegacyRedirectMode(@NonNull Activity activity) {
+        String generatedRedirectUri = activity.getPackageName().concat(".uberauth://redirect");
+        String setRedirectUri = sessionConfiguration.getRedirectUri();
+
+        if (redirectForAuthorizationCode) {
+            String message = "The Uber Authentication Flow for the Authorization Code Flow has "
+                    + "been upgraded in 0.8.0 and a redirect URI must now be supplied to the application. "
+                    + "You are seeing this error because the use of deprecated method "
+                    + "LoginManager.setRedirectForAuthorizationCode() indicates your flow may not "
+                    + "support the recent changes. See https://github"
+                    + ".com/uber/rides-android-sdk#authentication-migration-version"
+                    + "-08-and-above for resolution steps"
+                    + "to insure your setup is correct and then migrate to the non-deprecate "
+                    + "method LoginManager.setAuthCodeFlowEnabled()";
+
+            logOrError(activity, new IllegalStateException(message));
+            return true;
+        }
+
+        if (sessionConfiguration.getRedirectUri() == null) {
+            String message = "Redirect URI must be set in "
+                    + "Session Configuration.";
+
+            logOrError(activity, new NullPointerException(message));
+            return true;
+        }
+
+        if (!generatedRedirectUri.equals(setRedirectUri) &&
```

Comment:
The idea here is that we want to check if a developer previously had a redirect URI set from an earlier version of the SDK, and with the latest version, we add the generated scheme com.example.uberauth://redirect, but they have not updated the SessionConfig, which indicates they probably haven't registered the new one in the Uber Developer Dashboard.